packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { filter, find, includes, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	NavigableMenu,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as editorStore } from '@wordpress/editor';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+
+const DEFAULT_TEMPLATE_SLUGS = [
+	'front-page',
+	'single-post',
+	'page',
+	'archive',
+	'search',
+	'404',
+	'index',
+];
+
+export default function NewTemplate( { postType } ) {
+	const { templates, defaultTemplateTypes } = useSelect(
+		( select ) => ( {
+			templates: select( coreStore ).getEntityRecords(
+				'postType',
+				'wp_template'
+			),
+			defaultTemplateTypes: select(
+				editorStore
+			).__experimentalGetDefaultTemplateTypes(),
+		} ),
+		[]
+	);
+	const { addTemplate } = useDispatch( editSiteStore );
+
+	async function createTemplate( slug ) {
+		const { title, description } = find( defaultTemplateTypes, { slug } );
+
+		const { templateId } = await addTemplate( {
```

Comment:
Same here with the idea of using apiFetch to create the template. It would avoid the table updating in the background.